src/account.rs

```diff
@@ -16,21 +17,32 @@ pub(crate) fn print_account_address(path: Option<String>, account_index: usize)
     Ok(())
 }
 
-pub(crate) fn new_account(path: Option<String>) -> Result<()> {
-    let vault_path = handle_vault_path_argument(path)?;
-    let existing_accounts = Accounts::from_dir(&vault_path)?;
+fn new_account<P>(vault_path: P, password: &str) -> Result<WalletUnlocked>
```

Comment:
For the same reasons mentioned in my other comment on generic path arguments, I'd recommend just taking a `PathBuf` directly here if that's what we need and calling `.into()` on the caller side if necessary.